### PR TITLE
Scenes: Add support for Datasource variables

### PR DESCRIPTION
**Pull request by ivanortegaalba**
TODO:
- [x] Add unit testing for variables

**What is this feature?**

Add support for Datasource variables
Captura de pantalla 2022-11-22 a las 17 18 10

Relates to #58289

### CODE REVIEW

File: public/app/features/scenes/variables/variants/DataSourceVariable.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import { VariableDependencyConfig } from '../VariableDependencyConfig';
+import { VariableValueSelect } from '../components/VariableValueSelect';
+import { VariableValueOption } from '../types';
+
+import { MultiValueVariable, MultiValueVariableState, VariableGetOptionsArgs } from './MultiValueVariable';
+
+export interface DataSourceVariableState extends MultiValueVariableState {
+  query: string;
+  regex: string;
+  refresh: VariableRefresh;
```
Comment (torkelo) [inline public/app/features/scenes/variables/variants/DataSourceVariable.tsx:18]: don't think we need this option, this type of variable should always refresh on dashboard load

Comment (ivanortegaalba) [reply public/app/features/scenes/variables/variants/DataSourceVariable.tsx]: removed!

File: public/app/features/scenes/core/sceneGraph.ts [hunk 2]
```diff
@@ -89,13 +89,18 @@ export function getLayout(scene: SceneObject): SceneObject<SceneLayoutState> {
 /**
  * Interpolates the given string using the current scene object as context.   *
  */
-export function interpolate(sceneObject: SceneObject, value: string | undefined | null): string {
+export function interpolate(
```
Comment (ivanortegaalba) [inline public/app/features/scenes/core/sceneGraph.ts:92]: Missing unit tests here

Comment (ivanortegaalba) [reply public/app/features/scenes/core/sceneGraph.ts:92]: Actually, the sceneInterpolator has them already! ✅

**Review by torkelo — APPROVED** [review]

**Pull request merged**